src/SonarScanner.MSBuild.Common/RegexExtensions.cs

```diff
@@ -0,0 +1,118 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace SonarScanner.MSBuild.Common;
+
+public static class RegexExtensions
+{
+#pragma warning disable T0004 // T0004 Use 'SafeRegex.Matches' instead.
```

Comment:
I'm not sure what the benefit of the in-code suppression is. Instead of getting issues for `SafeRegex.Matches, ` we get issues for suppression. We can just accept them, and the file will be even closer to the one from the sonar-dotnet.